### ISSUE TITLE
Unable to find installation candidates for torch (2.0.1+cpu)

### ISSUE THREAD

**Issue by maxwkut** [iusztinpaul/hands-on-llms#58]
<img width="1897" alt="Screenshot 2023-12-01 at 3 22 14 PM" src="https://github.com/iusztinpaul/hands-on-llms/assets/52679528/0e6990a8-8dbf-4ba5-b748-850b3f24e910">

Great course so far! 

I'm having some trouble running `make install` in the streaming_pipeline folder. Any suggestions?

Main dependencies are all installed.

**Comment by iusztinpaul** [comment]
Thanks! Happy to hear that.

What OS are you using?

**Comment by maxwkut** [comment]
I'm using macOS Sonoma 14. Not sure if this is a mac specific problem though - in the video tutorial, it looked like @Paulescu ran the code on a mac as well and didn't have any issues.

**Comment by iusztinpaul** [comment]
Yeah. The streaming pipeline (out of the 3 modules) should work on macos. I tested my code only on Ubuntu. @Paulescu, do you know about this issue? Have you encountered it?

**Comment by Paulescu** [comment]
@maxwkut Have you tried removing the `pip install torch...` from the makefile and running `poetry add torch` ?

**Comment by iusztinpaul** [comment]
@maxwkut Have you managed to make it work?

**Comment by maxwkut** [comment]
@Paulescu @iusztinpaul I took out the `torch-cpu` dependency from `pyproject.toml` and then added the regular torch package using `poetry add torch`. Not sure if `torch` is a valid substitute for `torch-cpu`, but that at least allowed me to proceed to the next step: `make run_real_time`. Here I am now getting an error that there is an incompatible architecture (have arm64, but binary was compiled for x86_64)
<img width="1316" alt="Screenshot 2023-12-18 at 12 09 00 PM" src="https://github.com/iusztinpaul/hands-on-llms/assets/52679528/3e8154bb-8bae-46e3-a1d8-fd65e7dd0e3e">

**Comment by iusztinpaul** [comment]
Try to remove the torch completely from `pyproject.toml` by running `poetry remove torch` and then install it in your Poetry virtual environment using pip: `poetry run pip install torch==2.0.1` 

Let me know if this works. 

If that doesn't work, try to install it directly from the wheel, depending on your platform, something like this:
```
poetry add "https://download.pytorch.org/whl/cu118/torch-2.0.0%2Bcu118-cp310-cp310-linux_x86_64.whl" --platform linux
poetry add "https://download.pytorch.org/whl/cu118/torch-2.0.0%2Bcu118-cp310-cp310-linux_x86_64.whl" --platform darwin
```

**Comment by maxwkut** [comment]
As you can see if ran `poetry run pip install torch == 2.0.1` after removing it.

<img width="1913" alt="Screenshot 2023-12-18 at 5 39 37 PM" src="https://github.com/iusztinpaul/hands-on-llms/assets/52679528/e30ac1d9-0e5f-4b56-ba14-b92263c88e19">

It's saying that streaming-pipeline 0.1.0 requires torch<3.0.0,>=2.1.2?

**Comment by iusztinpaul** [comment]
Have you tried the second option?
